diff -Nur docker-1.11.2.orig/contrib/init/systemd/docker.service docker-1.11.2/contrib/init/systemd/docker.service
--- docker-1.11.2.orig/contrib/init/systemd/docker.service	2016-06-05 08:59:41.748796379 +0800
+++ docker-1.11.2/contrib/init/systemd/docker.service	2016-06-05 09:01:50.920346376 +0800
@@ -9,7 +9,8 @@
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/docker daemon -H fd://
+EnvironmentFile=/etc/conf.d/docker
+ExecStart=/usr/bin/docker daemon -H fd:// $DOCKER_OPTS
 MountFlags=slave
 LimitNOFILE=1048576
 LimitNPROC=1048576
