diff -Nur docker-1.11.2.orig/libcontainerd/remote_linux.go docker-1.11.2/libcontainerd/remote_linux.go
--- docker-1.11.2.orig/libcontainerd/remote_linux.go	2016-06-05 08:52:08.353779447 +0800
+++ docker-1.11.2/libcontainerd/remote_linux.go	2016-06-05 08:53:31.614782556 +0800
@@ -30,7 +30,7 @@
 	maxConnectionRetryCount   = 3
 	connectionRetryDelay      = 3 * time.Second
 	containerdShutdownTimeout = 15 * time.Second
-	containerdBinary          = "docker-containerd"
+	containerdBinary          = "containerd"
 	containerdPidFilename     = "docker-containerd.pid"
 	containerdSockFilename    = "docker-containerd.sock"
 	eventTimestampFilename    = "event.ts"
@@ -349,7 +349,7 @@
 	}
 
 	// Start a new instance
-	args := []string{"-l", r.rpcAddr, "--runtime", "docker-runc", "--start-timeout", "2m"}
+	args := []string{"-l", r.rpcAddr, "--runtime", "runc", "--start-timeout", "2m"}
 	if r.debugLog {
 		args = append(args, "--debug", "--metrics-interval=0")
 	}
