diff -Nu unzip-5.52.orig/unzpriv.h unzip-5.52/unzpriv.h
--- unzip-5.52.orig/unzpriv.h	2005-02-04 07:26:42.000000000 +0800
+++ unzip-5.52/unzpriv.h	2009-09-15 15:13:34.000000000 +0800
@@ -2472,7 +2472,7 @@
 #    endif
 #    define _OEM_INTERN(str1) {register uch *p;\
        for (p=(uch *)(str1); *p; p++)\
-         *p = native((*p & 0x80) ? oem2iso[*p & 0x7f] : *p);}
+	*p = native(*p);}
 #  endif
 #endif
 
