--- /usr/portage/official/sys-libs/glibc/glibc-2.20-r2.ebuild	2015-03-23 05:01:19.000000000 +0800
+++ glibc-2.20-r2.ebuild	2015-03-23 13:16:18.384382433 +0800
@@ -158,6 +158,12 @@
 }
 
 eblit-src_prepare-post() {
+	#scsi euctw patch
+	cp ${FILESDIR}/EUC-TW ${S}/localedata/charmaps || die "copy EUC-TW fail."
+	cp ${FILESDIR}/euc-tw.c ${S}/iconvdata || die "copy euc-tw.c fail."
+	cp ${FILESDIR}/euc2ucs.h ${S}/iconvdata || die "copy euc2ucs.h fail."
+	cp ${FILESDIR}/ucs2euc.h ${S}/iconvdata || die "copy ucs2euc.h fail."
+
 	cd "${S}"
 
 	if use hardened ; then
